extensions/opentelemetry/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/OpenTelemetryProcessor.java

```diff
@@ -29,6 +30,7 @@
 import io.quarkus.opentelemetry.runtime.OpenTelemetryRecorder;
 import io.quarkus.opentelemetry.runtime.QuarkusContextStorage;
 import io.quarkus.opentelemetry.runtime.tracing.cdi.WithSpanInterceptor;
+import io.quarkus.opentelemetry.runtime.tracing.restclient.OpenTelemetryClientFilter;
```

Comment:
watch this import -- depending on what it imports, you may run into trouble with a project w/o this enabled.

Supporting optional deps is why most of the telemetry/metrics support works with strings, not with imported class.getName()

Reply:
This is only used here:

https://github.com/radcortez/quarkus/blob/2b148a3d35cdc9aa52c84d57440a097a36ffd422/extensions/opentelemetry/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/OpenTelemetryProcessor.java#L113-L118

To add it as an additional class to index if the REST Client is Available, by testing this:

https://github.com/radcortez/quarkus/blob/2b148a3d35cdc9aa52c84d57440a097a36ffd422/extensions/opentelemetry/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/OpenTelemetryProcessor.java#L48-L58

The `OpenTelemetryClientFilter` is then added via CDI and `@Provider`:

https://github.com/radcortez/quarkus/blob/2b148a3d35cdc9aa52c84d57440a097a36ffd422/extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/restclient/OpenTelemetryClientFilter.java#L35-L36

I think it is fairly safe, but let me add some tests for it.